src/main.ts

```diff
@@ -92,9 +101,29 @@ export class OrientationVector {
     this.#vec.set(ov.x, ov.y, ov.z).normalize()
     this.th = ov.th
 
+    this.#onChangeCallback?.()
+
     return this
   }
 
+  fromArray(array: number[], offset = 0) {
+    this.#vec.set(array[offset]!, array[offset + 1]!, array[offset + 2]!)
+    this.th = array[offset + 3]!;
+
+    this.#onChangeCallback?.();
+
+    return this;
+  }
+
+  toArray(array: number[] = [], offset = 0) {
```

Comment:
Other toArray methods don't tend to modify a reference. Could this return a new array instead?

Reply:
In normal JS land I'd agree, but there are a few points that make me want to keep this function signature:
* It matches the API of [quaternion](https://github.com/mrdoob/three.js/blob/master/src/math/Quaternion.js#L635). I'd like this class to have as much parity with other THREE.js rotation classes as possible.
* This function signature is meant to be garbage collection friendly. Most THREE methods offer optional parameters for including pooled variables so that GC events aren't taking up a lot of CPU space for each frame.